### PR TITLE
Add coinWithBalanceIntent

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by plam-ml — APPROVED** [review]

File: sdk/typescript/src/transactions/blockData/v2.ts [hunk 1]
```diff
@@ -95,13 +98,17 @@ export const Argument = transform(
...
+	| { IntentResult: number }
+	| { NestedIntentResult: [number, number] },
 	// Output
 	| { $kind: 'GasCoin'; GasCoin: true }
 	| { $kind: 'Input'; Input: number; type?: 'pure' }
 	| { $kind: 'Input'; Input: number; type?: 'object' }
 	| { $kind: 'Result'; Result: number }
 	| { $kind: 'NestedResult'; NestedResult: [number, number] }
+	| { $kind: 'IntentResult'; IntentResult: number }
+	| { $kind: 'NestedIntentResult'; NestedIntentResult: [number, number] }
```
Comment (hayes-mysten) [inline sdk/typescript/src/transactions/blockData/v2.ts:111]: I think these might not be needed, I will probably just remove them and use normal Result and NestedResults unless we run into a usecase where there is a good reason to differentiate them

File: apps/wallet/src/ui/app/QredoSigner.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export class QredoSigner extends WalletSigner {
 
 	signMessage: WalletSigner['signMessage'] = async (input, clientIdentifier) => {
 		const signature = await this.signData(
-			messageWithIntent(IntentScope.PersonalMessage, input.message),
+			messageWithIntent('PersonalMessage', input.message),
```
Comment (hayes-mysten) [inline apps/wallet/src/ui/app/QredoSigner.ts:85]: I'm not 100% sure about this change.  I want to remove the duplicate intent types (bcs vs the ones in cryptography)

File: sdk/typescript/src/transactions/ObjectCache.ts [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+						}
+					}
+				}),
+			);
+
+			await next();
+
+			await Promise.all(
+				blockData.transactions.map(async (tx) => {
+					if (tx.MoveCall?.argumentTypes) {
```
Comment (hayes-mysten) [inline sdk/typescript/src/transactions/ObjectCache.ts:204]: we should probably deduplicate the move definitions here

File: sdk/typescript/src/transactions/ObjectCache.ts [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+						});
+
+						if (def) {
+							tx.MoveCall.argumentTypes = def.parameters;
+						}
+					}
+				}),
+			);
+
+			await next();
```
Comment (hayes-mysten) [inline sdk/typescript/src/transactions/ObjectCache.ts:200]: we aren't caching objects after resolution yet, just using the effects, but it's easy enough to do if we want that

File: sdk/typescript/src/transactions/ObjectCache.ts [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+	}) {
+		this.#client = client;
+		this.cache = new ObjectCache(options);
+	}
+
+	/**
+	 * Clears all Owned objects
+	 * Immutable objects, Shared objects, and Move function definitions will be preserved
+	 */
+	async reset() {
```
Comment (hayes-mysten) [inline sdk/typescript/src/transactions/ObjectCache.ts:283]: Not sure this is a good name.  The idea is to clear things that might have changed due to a transaction that happened outside the executor (just ownedObjects for now)

**Review by Jordan-Mysten — APPROVED** [review]

**Pull request merged**